lightkurve/correctors.py

```diff
@@ -487,3 +491,212 @@ def breakpoints(self, campaign):
+        tau = 30.
+
+        # set up gaussian process
+        kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
+        gp = celerite.GP(kernel)
+        sigma = gp.get_matrix(M(self.time)) + \
+                np.diag(np.sum(M(flux_err_crop).reshape(len(M(flux_err_crop)),
+                        -1), axis=1)**2)
+
+        # compute
+        A = np.dot(MX.T, np.linalg.solve(sigma, MX))
+        B = np.dot(MX.T, np.linalg.solve(sigma, M(rawflux)))
+        C = np.linalg.solve(A, B)
+
+        # compute detrended light curve
+        model = np.dot(X, C)
+        self.detrended_flux = rawflux - model + np.nanmean(rawflux)
+
+        # calculate errors (ignore warnings related to zero or negative errors)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            flux_err = np.nansum(self.flux_err[:, aperture_mask]**2, axis=1)**0.5
+
+        # estimate centroids
+        centroid_col, centroid_row = self.tpf.estimate_centroids(aperture)
+
+        # check type of input TPF and return corresponding LightCurve object
+        if isinstance(self.tpf, KeplerTargetPixelFile):
+            keys = {'centroid_col': centroid_col,
+                    'centroid_row': centroid_row,
+                    'quality': self.tpf.quality,
+                    'channel': self.tpf.channel,
+                    'campaign': self.tpf.campaign,
+                    'quarter': self.tpf.quarter,
+                    'mission': self.tpf.mission,
+                    'cadenceno': self.tpf.cadenceno,
+                    'ra': self.tpf.ra,
+                    'dec': self.tpf.dec,
+                    'label': self.tpf.header['OBJECT'],
+                    'targetid': self.tpf.targetid}
+            return KeplerLightCurve(time=self.time, flux=self.detrended_flux,
+                                    flux_err=flux_err, **keys)
+
+        elif isinstance(self.tpf, TessTargetPixelFile):
+            keys = {'centroid_col': centroid_col,
+                    'centroid_row': centroid_row,
+                    'quality': self.quality,
+                    'sector': self.sector,
+                    'camera': self.camera,
+                    'ccd': self.ccd,
+                    'cadenceno': self.cadenceno,
+                    'ra': self.ra,
+                    'dec': self.dec,
+                    'label': self.get_keyword('OBJECT'),
+                    'targetid': self.targetid}
+            return TessLightCurve(time=self.time, flux=self.detrended_flux,
+                                  flux_err=flux_err, **keys)
+
+        else:
+            warnings.warn("Input TargetPixelFile not recognized as a Kepler "
+                          "or TESS observation.", LightkurveWarning)
+            return LightCurve(time=self.time, flux=self.detrended_flux,
+                              flux_err=flux_err)
+
+    def diagnose(self, ax=None):
+        """Plot a figure to diagnose performance of PLD correction."""
```

Comment:
I feel like this should have some more interesting diagnostics than just plotting the results, but I don't have a good intuition for what they would be.